### PR TITLE
Pybito Tweaks for GP Work

### LINKED ISSUE #413: Tweaks to enable GP work
From @junseonghwan :

1. Currently, GP is lacking a feature to retrieve the branch lengths programmatically via pybito. The workaround that I have been using is to output the branch lengths and PCSPs to CSV and then reading it in from Python. Ideally, I would like to retrieve GP branch lengths from python in similarly to GatherBranchLengths — but instead of returning (index, vector<double>) map, (index, double) map.
2.To deal with the branch length estimates involving the outgroup (rootsplit), we will need to ask if a PCSP is a rootsplit.

### PR DESCRIPTION
**Pull request by davidrich27**
## Description

Exposes additional GP functions to pybito.
- Adds `get_edge_idx_to_pcsp_map`: returns a dictionary of `edge_id -> pcsp_bitset`
- Add `get_branch_lengths`: returns  vector of branch lengths, ordered by `edge_id`.

Closes #413

## Tests

Tested in `test_gp_tweaks` in `test/test_bito.py`

## Checklist:

* [x] Code follows our detailed [contribution guidelines](CONTRIBUTING.md)
* [x] `clang-format` has been run
* [x] TODOs have been eliminated from the code
* [x] Comments are up to date, document intent, and there are no commented-out code blocks
* [x] GitHub CI build on PR branch completed successfully

### CODE REVIEW

**Comment by junseonghwan** [comment]
Looks good to me. Thanks for implementing this.

File: src/subsplit_dag.cpp [hunk 1]
```diff
@@ -252,6 +252,18 @@ BitsetSizeMap SubsplitDAG::BuildEdgeIndexer() const {
   return edge_indexer;
 }
 
+SizeBoolVectorMap SubsplitDAG::BuildEdgeIdxToPCSPMap() const {
```
Comment (matsen) [inline src/subsplit_dag.cpp:255]: By the name I'd expect this to take values in PCSP Bitsets, and here it's returning vector bool. I see that you're doing that so you can get it from Python, but perhaps changing the name a bit?

**Review by matsen — CHANGES_REQUESTED** [review]

File: src/pybito.cpp [hunk 2]
```diff
@@ -432,14 +432,19 @@ PYBIND11_MODULE(bito, m) {
...
           "export_trees_with_a_pcsp", &GPInstance::ExportTreesWithAPCSP,
           R"raw(Write out trees with a given PCSP string to a Newick file (using current
           GP branch lengths).)raw",
           py::arg("pcsp_string"), py::arg("newick_path"))
       .def("subsplit_dag_to_dot", &GPInstance::SubsplitDAGToDot,
            R"raw(Write the current subsplit DAG to a DOT format file.)raw")
+      .def("get_branch_lengths", &GPInstance::GetBranchLengths,
+           "Return branch lengths from the GPInstance.")
+      .def(
+          "get_edge_idx_to_pcsp_map",
```
[the remark this answers is not in the document]
Comment (davidrich27) [reply src/pybito.cpp]: Changed.

File: src/subsplit_dag.cpp (hunk 1 above)
Comment (davidrich27) [reply src/subsplit_dag.cpp]: Does `BuildEdgeIdxToPCSPBoolVectorMap` work?

File: test/test_bito.py [hunk 3]
```diff
@@ -3,6 +3,8 @@
 If you want to see the results of the print statements, use `pytest -s`.
 """
 
+from asyncio import gather
```
[the remark this answers is not in the document]
Comment (davidrich27) [reply test/test_bito.py]: Okay, fixed.  And PR is ready for review if you haven't looked over everything yet.

**Review by matsen — APPROVED** [review]

**Pull request merged**